inspirehep/modules/theme/templates/inspirehep_theme/format/record/Inspire_HTML_detailed_macros.tpl

```diff
@@ -367,16 +367,16 @@
           <div class="col-md-2 hidden-sm hidden-xs" id="slider-thumbs">
             <!-- Left switcher of slider -->
             <ul class="hide-bullets">
-              {% set count = 0 %}
+              {% set count = [0] %}
               {% for url in record.urls %}
                  {% if url.value %}
                   {% if url.value.endswith(".png") or url.value.endswith(".jpg") %}
                     <li class="col-sm-12 show-plots-thumbnails">
-                      <a class="thumbnail" id="carousel-selector-{{ count }}">
+                      <a class="thumbnail" id="carousel-selector-{{ count[0] }}">
                         <img width="100" height="100" src="{{ url.value }}">
                       </a>
                     </li>
-                    {% set count = count + 1 %}
+                    {% if count.append(count.pop() + 1) %}{% endif %}
```

Comment:
The documentation of Jinja2 gives a better solution for this: http://jinja.pocoo.org/docs/2.9/templates/#for

> Inside of a for-loop block, you can access some special variables:
>
> | Variable | Description |
> | --- | --- |
> | loop.index0 | The current iteration of the loop. (0 indexed) |